src/server/bg_services/log_replication_scanner.js

```diff
@@ -122,93 +122,111 @@ class LogReplicationScanner {
     async process_candidates(src_bucket, dst_bucket, candidates) {
         const src_dst_objects_list = await this.head_objects(src_bucket.name, dst_bucket.name, candidates);
 
+        dbg.log1('log_replication_scanner: process_candidates src_dst_objects_list: ', src_dst_objects_list);
+
+        const copy_keys = [];
+        const delete_keys = [];
+
         for (const candidate of src_dst_objects_list) {
-            await this.process_candidate(src_bucket, dst_bucket, candidate);
+            const action = await this.process_candidate(candidate);
```

Comment:
unless I missed anything, `process_candidate` doesn't need to be async and there is no reason to wait for it.

Reply:
yes right, no need to async updated the code. Thanks